river/ensemble/streaming_random_patches.py

```diff
@@ -0,0 +1,478 @@
+        # TODO Find a way to verify if the model natively supports sample_weight
+        for _ in range(sample_weight):
+            self.model.learn_one(x=x_subset, y=y)
+
+        if self._background_learner:
+            # Train the background learner
+            # Note: Pass the original instance x so features are correctly
+            # selected based on the corresponding subspace
+            self._background_learner.learn_one(x=x, y=y, sample_weight=sample_weight,
+                                               n_samples_seen=n_samples_seen, rng=rng)
+
+        if not self.disable_drift_detector and not self.is_background_learner:
+            correctly_classifies = self.model.predict_one(x_subset) == y
+            # Check for warnings only if the background learner is active
+            if not self.disable_background_learner:
+                # Update the warning detection method
+                self.warning_detector.update(int(not correctly_classifies))
+                # Check if there was a change
+                if self.warning_detector.change_detected:
+                    self.n_warnings_detected += 1
+                    self._trigger_warning(
+                        all_features=all_features, n_samples_seen=n_samples_seen, rng=rng
+                    )
+
+            # ===== Drift detection =====
+            # Update the drift detection method
+            self.drift_detector.update(int(not correctly_classifies))
+            # Check if the was a change
+            if self.drift_detector.change_detected:
+                self.n_drifts_detected += 1
+                # There was a change, reset the model
+                self.reset(all_features=all_features, n_samples_seen=n_samples_seen, rng=rng)
+
+    def predict_proba_one(self, x):
+        # Select the features to use
+        x_subset = {k: x[k] for k in self.features} if self.features is not None else x
+
+        return self.model.predict_proba_one(x_subset)
+
+    def _trigger_warning(self, all_features, n_samples_seen: int, rng: np.random.RandomState):
+        # Randomly generate a new subspace from all the original features
+        subspace = None if self.features is None else random_subspace(
+            all_features=all_features, k=len(self.features), rng=rng
+        )
+
+        # Initialize the background learner
+        self._background_learner = self._background_learner_class(
+            idx_original=self.idx_original,
+            base_model=self.base_model,
+            metric=copy.deepcopy(self.metric),
+            created_on=n_samples_seen,
+            disable_background_learner=self.disable_background_learner,
+            base_drift_detector=self.base_drift_detector,
+            base_warning_detector=self.base_warning_detector,
+            is_background_learner=True,
+            features=subspace,
+            nominal_attributes=self.nominal_attributes,
+            rng=self._rng
+        )
+        # TODO Hard-reset the warning method
```

Comment:
Is this still valid?

Reply:
Actually, I forgot to refactor tyhis. It is fixed in d0dc434f8247752e226d357231df628681ad33cf